### DISCUSSION TITLE
repeatable & subfields & custom_html

### DISCUSSION THREAD

**Discussion by Akalavan** [Laravel-Backpack/community-forum#1016] [Q&A (Help)]
There is a field with the type 'repeatable'. This field has the 'subfields' attribute. It contains a field with the type 'custom_html'.
Can I get a non-top-level object in the 'value' field, but an object through a relation?
```php
[
    'tab'       => trans($this->trans_prefix . 'tabs.' . 'messages'),
    'label'     => trans($this->trans_prefix . 'relationship.' . 'messages'),
    'name'      => 'messages',
    'type'      => 'repeatable',
    'subfields' => [
        [
            'label'   => trans($this->trans_prefix_messages . 'body'),
            'name'    => 'body_viewer',
            'type'    => 'custom_html',
            'value'   => function ($entity) {
               //
            },
            'wrapper' => [
                'class' => 'form-group col-lg-12 col-md-12 col-md-12',
            ],
        ],
  ],
```
In this example, the top-level object is "Ticket".
It has a "hasMany" relationship with the "Message" entity.
Can I get in "custom_html" in "value" an object of type "Message" and not "Ticket"

**Comment by jcastroa87** [comment]
Hello @Akalavan 

Im sorry I don't fully understand your request, could you elaborate the question further and indicate what you tried, the code and if you have any errors.

Thanks.

**Reply by Akalavan** [reply]
Sorry I didn't have time to answer.
I have a "Ticket" model. The Ticket model has a connected "Messages" model.
```php
/**
* Attitude to messages 
* 
* @return HasMany
*/
public function messages(): HasMany
{
    return $this->hasMany(Messages::class);
}
```
In the controller of the "Ticket" model, I want to display entities associated with the "Messages" model.
To do this, use the following field
```php
[
    'tab'       => 'messages',
    'label'     => 'messages',
    'name'      => 'messages',
    'type'      => 'repeatable',
    'subfields' => [
        [
            'label'   => trans($this->trans_prefix_messages . 'body'),
            'name'    => 'body_viewer',
            'type'    => 'custom_html',
            'value'   => function ($entity) {
               //
            },
            'wrapper' => [
                'class' => 'form-group col-lg-12 col-md-12 col-md-12',
            ],
        ],
  ],
```
In the "custom_html" field, in the "value" attribute, the "Ticket" entity
```php
'value'   => function ($entity) {
//   $entity::class is "Ticket"
},
```
I want to get an entity of the "Message" type in the "custom_html" field, in the "value" attribute.
```php
'value'   => function ($entity) {
//   $entity::class is "Message"
},
```
Can I get an entity of the "Message" type. So that the $entity variable has a "Message" class model?
The field needs exactly "custom_html".

**Comment by jcastroa87** [comment]
Due to not activity i will close this issue, but please feel free to re-open or create a new one if needed.

Cheers.
